harmonica/tests/test_gradient_boosted_eqs.py

```diff
@@ -0,0 +1,235 @@
+    expected_required_memory = data_p_window * sources_p_window * itemsize
+    # Estimate required memory
+    required_memory = EquivalentSourcesGB.estimate_required_memory(
+        coordinates, window_size=window_size, dtype=dtype
+    )
+    assert required_memory == expected_required_memory
+
+
+# -----------------------------------------------------------------------
+# Test the fitting and predictions of gradient-boosted equivalent sources
+# -----------------------------------------------------------------------
+
+
+@pytest.mark.use_numba
+@pytest.mark.parametrize("weights", [None, np.ones((8, 8))])
+def test_gb_eqs_small_data(region, weights):
+    """
+    Check predictions against synthetic data using few data points for speed
+    """
+    _, _, coordinates, data = build_sample_sources_and_data(
+        region=region, coords_shape=(8, 8)
+    )
+    # The interpolation should be good enought on the data points
+    # Gradient-boosted equivalent sources don't perform well on small data, so
+    # we will check if the error is no larger than 1mGal
+    # (sample data ranges from approximately -7mGal to 7mGal)
+    eqs = EquivalentSourcesGB(depth=1e3, window_size=1e3, random_state=42)
+    eqs.fit(coordinates, data, weights=weights)
+    npt.assert_allclose(data, eqs.predict(coordinates), atol=1)
+
+
+@run_only_with_numba
+def test_gradient_boosted_eqs_single_window(region):
+    """
+    Test GB eq-sources with a single window that covers the whole region
+    """
+    points, masses, coordinates, data = build_sample_sources_and_data(region)
+    # The interpolation should be perfect on the data points
+    eqs = EquivalentSourcesGB(window_size=region[1] - region[0])
+    eqs.fit(coordinates, data)
+    npt.assert_allclose(data, eqs.predict(coordinates), rtol=1e-5)
+    # Gridding onto a denser grid should be reasonably accurate when compared
+    # to synthetic values
+    grid = vd.grid_coordinates(region=region, shape=(60, 60), extra_coords=0)
+    true = hm.point_mass_gravity(grid, points, masses, field="g_z")
+    npt.assert_allclose(true, eqs.predict(grid), rtol=1e-3)
+
+
+@run_only_with_numba
+def test_gradient_boosted_eqs_predictions(region):
+    """
+    Test GB eq-sources predictions
+    """
+    # Define a large region
+    points, masses, coordinates, data = build_sample_sources_and_data(region)
+    # The interpolation should be sufficiently accurate on the data points
+    eqs = EquivalentSourcesGB(window_size=1e3, depth=1e3, damping=None, random_state=42)
+    eqs.fit(coordinates, data)
+    npt.assert_allclose(data, eqs.predict(coordinates), atol=1e-2 * vd.maxabs(data))
+
+    # Gridding onto a denser grid should be reasonably accurate when compared
+    # to synthetic values
+    grid = vd.grid_coordinates(region=region, shape=(60, 60), extra_coords=0)
+    true = hm.point_mass_gravity(grid, points, masses, field="g_z")
+    npt.assert_allclose(true, eqs.predict(grid), atol=1e-2 * vd.maxabs(true))
```

Comment:
@leouieda pay attention to this test function. Does it make sense to compare with `atol`?